src/pid.cpp

```diff
@@ -299,15 +306,16 @@ double Pid::computeCommand(double error, ros::Duration dt)
 {
 
   if (dt == ros::Duration(0.0) || std::isnan(error) || std::isinf(error))
-    return 0.0;
+    return cmd_;
 
   double error_dot = d_error_;
 
   // Calculate the derivative error
-  if (dt.toSec() > 0.0)
+  if (dt.toSec() > 0.0 && valid_p_error_last_)
   {
     error_dot = (error - p_error_last_) / dt.toSec();
     p_error_last_ = error;
+    valid_p_error_last_ = true;
```

Comment:
I don't understand this change. This code will only be reached if `valid_p_error_last_` is already `true`.
If it is `false`, only `reset()` can set it to `true` again.

Reply:
My bad, I don't know how this slipped in this way.. Sorry
Added a test